src/util.cpp

```diff
@@ -579,6 +579,26 @@ const fs::path &GetDataDir(bool fNetSpecific)
     return path;
 }
 
+fs::path GetWalletDir()
+{
+    fs::path path;
+
+    if (gArgs.IsArgSet("-walletdir")) {
+        path = fs::system_complete(gArgs.GetArg("-walletdir", ""));
+        if (!fs::is_directory(path)) {
+            path = "";
```

Comment:
I don't love this. At the moment you're using `path = ""` to essentially return an error to the caller (since on start-up, the calling function checks that the returned `fs::path` is a directory). That's not at all clear from reading just this function.

Can you be more explicit about how the error is returned, either by explicitly returning bool or at the least by adding a comment here?